src/languages/clickhouse/clickhouse.formatter.ts

```diff
@@ -0,0 +1,161 @@
+import { DialectOptions } from '../../dialect.js';
+import { expandPhrases } from '../../expandPhrases.js';
+import { functions } from './clickhouse.functions.js';
+import { dataTypes, keywords, keywordPhrases } from './clickhouse.keywords.js';
+
+const reservedSelect = expandPhrases(['SELECT [DISTINCT]']);
+
+const reservedClauses = expandPhrases([
+  // https://clickhouse.com/docs/sql-reference/statements/explain
+  'EXPLAIN [AST | SYNTAX | QUERY TREE | PLAN | PIPELINE | ESTIMATE | TABLE OVERRIDE]',
+]);
+
+const standardOnelineClauses = expandPhrases([
+  // https://clickhouse.com/docs/sql-reference/statements/create
+  'CREATE [OR REPLACE] [TEMPORARY] TABLE [IF NOT EXISTS]',
+  // https://clickhouse.com/docs/sql-reference/statements/update
+  'UPDATE',
+  // https://clickhouse.com/docs/sql-reference/statements/system
+  'SYSTEM RELOAD {DICTIONARIES | DICTIONARY | FUNCTIONS | FUNCTION | ASYNCHRONOUS METRICS} [ON CLUSTER]',
+  'SYSTEM DROP {DNS CACHE | MARK CACHE | ICEBERG METADATA CACHE | TEXT INDEX DICTIONARY CACHE | TEXT INDEX HEADER CACHE | TEXT INDEX POSTINGS CACHE | REPLICA | DATABASE REPLICA | UNCOMPRESSED CACHE | COMPILED EXPRESSION CACHE | QUERY CONDITION CACHE | QUERY CACHE | FORMAT SCHEMA CACHE | DROP FILESYSTEM CACHE}',
```

Comment:
`SYSTEM DROP {... | DROP FILESYSTEM CACHE}`

will result in:

`SYSTEM DROP DROP FILESYSTEM CACHE`